geniza/corpus/management/commands/export_annotations.py

```diff
@@ -0,0 +1,143 @@
+import json
+import os
+from collections import defaultdict
+from urllib.parse import urlparse
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from geniza.annotations.models import Annotation, annotations_to_list
+from geniza.corpus.models import Document
+from geniza.footnotes.models import Footnote
+
+
+class Command(BaseCommand):
+    """Backup annotation data and synchronize to GitHub"""
+
+    v_normal = 1  # default verbosity
+
+    def handle(self, *args, **options):
+        if not getattr(settings, "ANNOTATION_BACKUP_PATH"):
+            raise CommandError(
+                "Please configure ANNOTATION_BACKUP_PATH in django settings"
+            )
+
+        annotations_output_dir = os.path.join(
+            settings.ANNOTATION_BACKUP_PATH, "annotations"
+        )
+        # define paths and ensure directories exist for compiled transcription
+        transcription_output_dir = {}
+        for output_format in ["txt", "html"]:
+            print([settings.ANNOTATION_BACKUP_PATH, "transcriptions", output_format])
+            format_path = os.path.join(
+                settings.ANNOTATION_BACKUP_PATH, "transcriptions", output_format
+            )
+            transcription_output_dir[output_format] = format_path
+            os.makedirs(format_path, exist_ok=True)
+
+        # identify content to backup based on documents with digital editions
+        docs = Document.objects.filter(
+            footnotes__doc_relation__contains=Footnote.DIGITAL_EDITION
+        ).distinct()
+        self.stdout.write(
+            "Backing up annotations for %d documents with digital editions"
+            % docs.count()
+        )
+        for document in docs[:3]:
+            print(document)
+            # use PGPID for annotation directory name
+            # path based on recommended uri pattern from the spec
+            # {prefix}/{identifier}/list/{name}
+            output_dir = os.path.join(annotations_output_dir, str(document.id), "list")
+            # ensure output directory exists
+            os.makedirs(output_dir, exist_ok=True)
+
+            # find all annotations for this document
+            annos = Annotation.objects.filter(
+                content__target__source__partOf__id=document.manifest_uri
```

Comment:
Same thought here as "documents with digital editions", but I do find I have to refer back to this query elsewhere in code every time I need it, so a convenience method (on `Document`?) could maybe be helpful.

Reply:
it does make sense on `Document` but that model and file are already enormous! is there a logical mixin we could create with annotation specific functionality? AnnotationTargetSourceMixin ? (ugh)

Reply:
Maybe just AnnotationTargetMixin? It's not semantically correct in terms of IIIF, but I think it could easily mean "the annotation's target is a part of this document" to PGP code contributors. OTOH, I'm not opposed to making the Document model slightly longer to add a small convenience method like this. But I do share your concern!